tests/nosetests/pyanaconda_tests/module_source_harddrive_test.py

```diff
@@ -122,23 +135,22 @@ def ready_state_test(self, ismount):
         ismount.return_value = True
 
         task = self.module.set_up_with_tasks()[0]
-        task.get_result = Mock(return_value=("/my/path", False))
+        task.get_result = Mock(return_value=SetupHardDriveResult("/my/path", ""))
         task.succeeded_signal.emit()
 
         self.assertEqual(self.module.get_state(), SourceState.READY)
         ismount.assert_called_once_with(self.module._device_mount)
 
-
     def return_handler_test(self):
         """Hard drive source setup result propagates back."""
         task = _create_setup_task()
         # Test only the returning. To do that, fake what the magic in start() does.
         # Do not run() the task at all, less mocking needed that way.
-        task._set_result((iso_mount_location, True))
+        task._set_result(SetupHardDriveResult(iso_mount_location, "iso_name.iso"))
```

Comment:
An explicit test for the inverse situation, where you'd call `_set_result` with the second string empty, is missing. I believe that's why you didn't catch the "always true" problem.  Coverage is 100%, so the code path is taken in some other test. 

Maybe you could add the assert for `is_iso_mounted==False` in `success_find_dir_test`? And then also for `is_iso_mounted==True` in `success_find_iso_test`.

Reply:
> An explicit test for the inverse situation, where you'd call _set_result with the second string empty, is missing. I believe that's why you didn't catch the "always true" problem. 

Good point I'll extend that test.

> Coverage is 100%, so the code path is taken in some other test.

Coverage is 100% because I've added this test part when fixing this.
https://github.com/rhinstaller/anaconda/pull/2503/files#diff-74093229cac143f46d63bb39fee26e08R98

> Maybe you could add the assert for is_iso_mounted==False in success_find_dir_test? And then also for is_iso_mounted==True in success_find_iso_test.

I'm not sure I follow you here. I can't check `is_iso_mounted` because these tests are working with task only. Module/Interface is not involved as far as I know.

Reply:
Sorry, my bad, I only skimmed that part.